The Migration Update (#124)
sklauncher/docs: docs/servers/lan-world.mdx

---
title: Open to LAN worlds
description: You can play in your Singleplayer world with up to 8 friends in LAN
sidebar_position: 1
---

You can play in your Singleplayer world with up to 8 friends in LAN, thanks to the fact that since Release 1.3.1 the game has an integrated server\* in the client.

<small>\* - If you're going to play on a version prior to the mentioned one, you'll have to create a [dedicated server](/servers/dedicated-server).</small>

## Hosting

> 1. You want to be the creator, so you start the game and go to *Singleplayer* -> *Play Selected World* / *Create New World* and once the world is loaded you can press <kbd>ESC</kbd> and click **Open to LAN**.
> 2. Now choose the *Game Mode* and if you want to *Allow Commands*, then click **Start LAN World**.

Reviewer: ```suggestion
1. You want to be the host, so you start the game, fire up the world of choice, press <kbd>ESC</kbd> and click **Open to LAN**.
2. Now configure the server settings to your liking, and then click **Start LAN World**.
```
No need to overcomplicate things.

3. Done, you should see a message like this in the chat: `Local game hosted on port [50000]` (the port may be different).
    - Depending on the version it can show you the IP and Port (for example: 192.168.1.100:50000).
4. Tell your friends to join.

## Joining

> 1. The host already did everything and you want to connect, then you start the game and go to *Multiplayer*.

Reviewer: ```suggestion
1. The host already did everything and you want to connect, so you start the game and go to *Multiplayer*.
```

## Common issues
There are many possible problems and some are detailed in [this](https://www.howtogeek.com/242375/how-to-troubleshoot-minecraft-lan-game-problems/) external link.\
Assuming the best case scenario where you know about IP, Firewall, etc. and therefore you've everything properly configured, there are 2 very common and "difficult" problems.

### Failed to login: Invalid session
That's already answered [here](/troubleshooting/game-related#failed-to-login-invalid-session), but you don't have access to the `server.properties` file in this case!\
There are two possible solutions, the 1st one would work in any case.

#### Vanilla (no mods)
Players will only be able to connect if it appears in Multiplayer under *Scanning for games on your local network*.\
If the world isn't listed there (since connecting by IP:Port will give the invalid session error), you'll have to do [this](#the-world-is-not-listed-in-the-lan-list).

#### Modding
There are some mods that only add the option to change the Online Mode and this is what you're looking for.\

> Honestly you should have already seen the [Modding](/category/modding) guide and have an idea that, depending on the mod you want to install, you'll need to install the required Mod Loader first.

Reviewer: ```suggestion
In case you have never installed mods before, you can learn how to do that in the [Modding](/category/modding) guide.
```
This sentence seems rude. I would change it.
